spec/models/carto/mapcap_spec.rb

```
@@ -104,29 +104,53 @@
       @ids_json = nil
-      @regenerated_visualization = nil
     end
 
     it 'should preserve map' do
-      @regenerated_visualization.map.id.should eq @map.id
+      regenerated_visualization = @mapcap.regenerate_visualization
+      regenerated_visualization.map.id.should eq @map.id
     end
 
     it 'should preserve visualization' do
-      @regenerated_visualization.id.should eq @visualization.id
+      regenerated_visualization = @mapcap.regenerate_visualization
+      regenerated_visualization.id.should eq @visualization.id
     end
 
     it 'should preserve user' do
-      @regenerated_visualization.user.id.should eq @visualization.user.id
+      regenerated_visualization = @mapcap.regenerate_visualization
+      regenerated_visualization.user.id.should eq @visualization.user.id
     end
 
     it 'should preserve permission' do
-      @regenerated_visualization.permission.id.should eq @visualization.permission.id
+      regenerated_visualization = @mapcap.regenerate_visualization
+      regenerated_visualization.permission.id.should eq @visualization.permission.id
+    end
+
+    describe 'without user DB' do
+      before(:all) do
+        @user_nodb = FactoryGirl.create(:carto_user, private_tables_enabled: true)
+        @map_nodb, @table_nodb, @table_visualization_nodb, @visualization_nodb = create_full_visualization(@user_nodb)
+        @mapcap_nodb = Carto::Mapcap.create!(visualization_id: @visualization_nodb.id)
+        @actual_db_name = @user_nodb.database_name
+        @user_nodb.update_attribute(:database_name, 'wadus')
+        @mapcap_nodb.reload
+      end
+
+      after(:all) do
+        @user_nodb.update_attribute(:database_name, @actual_db_name)
+        destroy_full_visualization(@map_nodb, @table_nodb, @table_visualization_nodb, @visualization_nodb)
+        @user_nodb.destroy
+      end
+
+      it 'should work' do
+        CartoDB::Logger.expects(:warning).never
+        @mapcap_nodb.regenerate_visualization
+      end
```

You're right. I'll add `User.any_instance.expects(:in_database).never`. 👍 

I'll keep the warning level as well, to be safer (there's a legit rescue swallowing errors, I prefer having a double check on this, if you don't mind).